src/server/auth/errors.ts

```diff
@@ -4,12 +4,15 @@ import { OAuthErrorResponse } from "../../shared/auth.js";
  * Base class for all OAuth errors
  */
 export class OAuthError extends Error {
+  static errorCode: string;
+  public errorCode: string;
```

Comment:
this and the `this.constructor` below seem odd, and I'm wondering if there's another way to do this.

Reply:
I've refactored this slightly, makes it more obvious where `errorCode` is actually coming from.